pkg/controller/nodes/task/handler.go

```diff
@@ -216,16 +220,30 @@ func (t *Handler) Setup(ctx context.Context, sCtx handler.SetupContext) error {
 			return regErrors.Wrapf(err, "failed to load plugin - %s", p.ID)
 		}
 		for _, tt := range p.RegisteredTaskTypes {
-			logger.Infof(ctx, "Plugin [%s] registered for TaskType [%s]", cp.GetID(), tt)
-			// TODO(katrogan): Make the default task plugin assignment more explicit (https://github.com/lyft/flyte/issues/516)
-			t.defaultPlugins[tt] = cp
+			for _, defaultTaskType := range p.DefaultForTaskTypes {
```

Comment:
so the value exists in the struct defined in plugins, but it is overwritten by propeller when reading from the config. i defined it in the plugin struct because that is what we're using to schlep around plugins in this resolution code but it's not meant to be assigned there. if it's preferable, i can create a separate struct in propeller, copy all the fields over and remove the DefaultForTaskTypes from the plugins struct